engine/src/main/java/io/zeebe/engine/processing/deployment/distribute/CompleteDeploymentDistributionProcessor.java

```diff
@@ -41,8 +46,13 @@ public void processRecord(
     stateWriter.appendFollowUpEvent(
         deploymentKey, DeploymentDistributionIntent.COMPLETED, record.getValue());
 
-    // todo(zell): https://github.com/zeebe-io/zeebe/issues/6173
-    // check whether it was the last pending
-    // then write Deployment.FULLY_DISTRIBUTED
+    if (!deploymentState.hasPendingDeploymentDistribution(deploymentKey)) {
+      final var storedDeploymentRecord = deploymentState.getStoredDeploymentRecord(deploymentKey);
+
+      if (storedDeploymentRecord != null) {
+        stateWriter.appendFollowUpEvent(
+            deploymentKey, DeploymentIntent.FULLY_DISTRIBUTED, storedDeploymentRecord);
```

Comment:
We should write a test to verify that the `FULLY_DISTRIBUTED` record is written.

Reply:
I will do when I wired all together. Currently we have no complete command which is written by the others so :man_shrugging: :D